### PR TITLE
Add initial tokenizer for Python

### PR DESCRIPTION
**Pull request by certik**
One can use it using `lpython --show-tokens somefile`. Currently the tokenizer is a Fortran tokenizer, so we need to adapt it to become a Python tokenizer.

Towards #298.

### CODE REVIEW

**Review by Thirumalai-Shaktivel — APPROVED** [review]
LGTM! Let's merge this once we fix the CI

**Comment by Thirumalai-Shaktivel** [comment]
Hi @certik, I updated the tokenizer to recognize Comment, DocString, Symbols and also made the CI pass.
Can you review this??

**Comment by Thirumalai-Shaktivel** [comment]
I tested this with many Python files. I got some errors, which I fixed and will push here.
The rest, `--show-tokens` provides output without any errors!
Now I think we can move to the parser (Bison).

**Comment by certik** [comment]
Can you get the CI passing again please?

File: run_tests.py
Comment (certik) [inline run_tests.py:82]: I think you forgot to update the prefix:
```suggestion
            run_test("tokens", "lpython --no-color --show-tokens {infile} -o {outfile}",
```

Comment (certik) [reply run_tests.py:82]: (You'll have to update the tests --- remove everything in `tests/reference/*` and rerun `./run_tests.py -u`.

**Comment by certik** [comment]
I left a fix above, after the update I think this is good to go in.

After it is merged, we should do two more steps before moving to the parser:

* Emit `indent` and `dedent` tokens properly (i.e., handle indentation in the tokenizer)
* Access the tokens from CPython somehow (figure out how to do that), and then compare against it, that we are getting exactly those tokens and nothing more or less.
* Remove all the Fortran stuff from the tokenizer

It doesn't have to be perfect, it won't be perfect, but I would definitely do the above three things. Then we can move to the parser and start parsing it. And then we'll iterate on the design of the tokenizer as needed.

File: run_tests.py
Comment (Thirumalai-Shaktivel) [reply run_tests.py]: Good catch, Thanks!

**Comment by Thirumalai-Shaktivel** [comment]
Perfect, I will work on `indent` and `dedent` and send a PR.
Thanks for the review!

**Comment by certik** [comment]
Thanks!  I think this looks good, now we just have to iteratively improve it, the TODO list is written up at: https://github.com/lcompilers/lpython/issues/298#issuecomment-1086156155

**Pull request merged**
